mne/io/base.py

```diff
@@ -33,17 +33,156 @@
 from ..parallel import parallel_func
 from ..utils import (_check_fname, _check_pandas_installed,
+                     _check_pandas_index_arguments,
                      check_fname, _get_stim_channel, object_hash,
-                     logger, verbose)
+                     logger, verbose, deprecated)
 from ..viz import plot_raw, plot_raw_psds, _mutable_defaults
 from ..externals.six import string_types
 from ..event import concatenate_events
 
 
+class ToDataFrameMixin(object):
+    '''Class to add to_data_frame capabilities to certain classes.'''
+    def to_data_frame(self, picks=None, index=None, scale_time=1e3,
+                      scalings=None, copy=True, start=None, stop=None):
+        """Get the epochs as Pandas DataFrame
+
+        Export epochs data in tabular structure with MEG channels as columns
+        and three additional info columns 'epoch', 'condition', and 'time'.
+        The format matches a long table format commonly used to represent
+        repeated measures in within-subject designs.
+
+        Parameters
+        ----------
+        picks : array-like of int | None
+            If None only MEG and EEG channels are kept
+            otherwise the channels indices in picks are kept.
+        index : tuple of str | None
+            Column to be used as index for the data. Valid string options
+            are 'epoch', 'time' and 'condition'. If None, all three info
+            columns will be included in the table as categorial data.
+        scale_time : float
+            Scaling to be applied to time units.
+        scalings : dict | None
+            Scaling to be applied to the channels picked. If None, defaults to
+            ``scalings=dict(eeg=1e6, grad=1e13, mag=1e15, misc=1.0)`.
+        copy : bool
+            If true, data will be copied. Else data may be modified in place.
+        start, stop : int | None
+            If it is a Raw object, this defines a start/stop index for creating
+            the dataframe from a slice. The times will be interpolated from the
+            index and the sampling rate of the signal.
+
+        Returns
+        -------
+        df : instance of pandas.core.DataFrame
+            A dataframe suitable for usage with other
+            statistical/plotting/analysis packages. Column/Index values will
+            depend on the object type being converted, but should be
+            human-readable.
+        """
+        from ..epochs import Epochs, EpochsArray
+        from ..evoked import Evoked, EvokedArray
+        from .fiff import RawFIFF
+        from .array import RawArray
+        from ..source_estimate import SourceEstimate
+
+        pd = _check_pandas_installed()
+
+        ####### ADDITION #######
+        if isinstance(self, (Epochs, EpochsArray)):
+            default_index = ['condition', 'epoch', 'time']
+        elif isinstance(self, (RawFIFF, RawArray, Evoked,
+                               EvokedArray, SourceEstimate)):
```

Comment:
and instead of doing multiple checks against types in multiple places, can you do them all here, once, and do all the necessary triaging so that the rest of the code can run without instance-style triaging?